src/app/services/keycloak.service.ts

```diff
@@ -174,6 +183,51 @@ export class KeycloakService {
    * @memberof KeycloakService
    */
   login(idpHint: string) {
-    return this.keycloakAuth && this.keycloakAuth.login({ idpHint: idpHint });
+    let redirectUri = window.location.href;
+    // by default keycloak login will want to redirect back to the login page
+    // redirect to '/dayuse' instead
+    if (redirectUri.endsWith('/login')) {
+      redirectUri = redirectUri.slice(0, redirectUri.lastIndexOf('/'));
+    }
+    return this.keycloakAuth && this.keycloakAuth.login({ idpHint: idpHint, redirectUri: redirectUri });
+  }
+
+  /**
+   * Infers the identity provider from the JWT token
+   *
+   * @memberof KeycloakService
+   */
+  getIdpFromToken(): string {
+    const token = this.getToken();
+
+    if (!token) {
+      return '';
+    }
+
+    const jwt = JwtUtil.decodeToken(token);
+
+    // idir users have an idir_userid property
+    if (jwt.idir_userid !== undefined) {
+      return this.idpHintEnum.IDIR;
+    }
+
+    // if extra steps are completed in the keycloak config
+    // bceid users will have a bceid_userid property
+    if (jwt.bceid_userid !== undefined) {
+      return this.idpHintEnum.BCEID;
+    }
+
+    const preferredUsernameParts = jwt.preferred_username.split('@');
+    if (preferredUsernameParts.length > 1) {
+      const domainPart = preferredUsernameParts[1].toLowerCase();
+      // the preferred_username for BCEID users is `username@bceid-basic-and-business`
+      if (domainPart.startsWith('bceid')) {
+        return this.idpHintEnum.BCEID;
+      }
+    }
```

Comment:
I'd prefer to leave it as a safety check in case somebody misses a step when configuring Keycloak in Dev/Test/Prod environments.  I just checked Attendance & Revenue, and it is also missing `idir_userid`, so I think I should add a secondary check for idir as well.

Reply:
@marklise could we make it fail less-silently when the mappers aren't configured in Keycloak?  e.g. maybe we could show a toast after a successful login that actually says something like:

**Login Successful**
Authenticated with BC Services Card

Reply:
I think the mappers should  be properly setup first, instead of having safety checks in code that are unnecessary after the fact.  I'm personally not a fan of popups/toasts on successful login.

Reply:
I set up the mappers on DEV for both DUP and A&R, and I removed the secondary checks